projects/core/src/model/product-search.model.ts

```diff
@@ -31,9 +31,22 @@ export interface Facet {
   multiSelect?: boolean;
   name?: string;
   priority?: number;
-  topValues?: FacetValue[];
   values?: FacetValue[];
   visible?: boolean;
+
+  /**
+   * Indicates the top values that will be shown instantly. The top values can be
+   * controlled by business users per facet.
+   */
+  topValueCount: number;
+
+  /**
+   * The OCC backend has topValues with duplicated facet data.
+   * This is not used in the UI, but normalized in the `topValueCount` property.
+   *
+   * TODO: remove the `topValues` property from the UI model.
```

Comment:
Is this a TODO for this ticket?

Reply:
We will do this in the future when we really decouple UI model from OCC backend model. I'll optimize the TODO